### PR TITLE
Is1233/auto update postgres

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] adds a postgres migration docker container for use in local machine when initializing the database, or upgrading the database
- [x] adds a Makefile recipe to upgrade the database from the command line ```make postgres-upgrade```

Behind the scene, the container has the **sc-pg** package from github (current branch, not LOCAL changes). then the service is attached to the simcore network, and by default runs ```sc-pg upgrade```

## Related issue number

#1233 
@ignapas please check what else is necessary. I think this closes the issue as far as I know.

## How to test

```
make leave # ensure nothing is running
docker volume prune # ensure the database volume is removed from host
make build # ensure the latest and greatest is available
make up-prod # runs the latest build
# now have a look in [http://127.0.0.1:18080](http://127.0.0.1:18080), the database is empty
make postgres-upgrade # will build the migration container and run it straight away
# now have a look again in [http://127.0.0.1:18080](http://127.0.0.1:18080), the database is initialized/upgraded
```
## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/postgres/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM python:3.6-alpine
```
Comment (pcrespov) [inline services/postgres/Dockerfile]: I think the folder shall NOT be called ``postgres`` . It gives the impression that this is to build the postgres service and it is not!
Please rename ``service/postgres`` to e.g. ``postgres-migrator`` or ``pg-migrator``  instead

File: services/postgres/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    git \
+    python3-dev \
+    musl-dev \
+    postgresql-dev \
+    && git clone --single-branch --branch ${GIT_BRANCH} ${GIT_REPOSITORY} \
+    && pip install osparc-simcore/packages/postgres-database[migration] \
+    && apk del --no-cache .build-deps
+
+# runtime dependency
+RUN apk add --no-cache --virtual .build-deps \
```
Comment (pcrespov) [inline services/postgres/Dockerfile]: MNIOR: need to add --virtual? even if not deleted?

File: services/postgres/Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
+.DEFAULT_GOAL := help
+
+REPO_BASE_DIR  = $(abspath $(CURDIR)/../../../)
+VENV_DIR ?= $(abspath $(REPO_BASE_DIR)/.venv)
+
+APP_NAME     := $(notdir $(CURDIR))
```
Comment (pcrespov) [inline services/postgres/Makefile]: As an devop convention, the name of the folder is the name of the service you run ... this supports the idea of renaming folder to something other than ``postgres`` :-)

File: services/postgres/Makefile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+# this is a way to get a valid git repo/git branch
+LOCAL_BRANCH = $(shell git name-rev --name-only HEAD)
```
Comment (pcrespov) [inline services/postgres/Makefile]: nice!!

File: services/postgres/Dockerfile (hunk 1 above)
Comment (sanderegg) [reply services/postgres/Dockerfile]: you are right and it is not a service but the postgres-database package packaged in a docker container..
so I moved it there.

**Pull request merged**